pkg/controller/configuration/config_template_test.go

```diff
@@ -366,16 +366,31 @@ true
 
 			// for small instance class
 			Expect(calMysqlPoolSizeByResource(&ResourceDefinition{
-				MemorySize: 1024 * 1024 * 1024,
+				MemorySize: 1024 * 1024 * 0.5,
 				CoreNum:    1,
 			}, false)).Should(Equal("128M"))
 
+			Expect(calMysqlPoolSizeByResource(&ResourceDefinition{
+				MemorySize: 1024 * 1024 * 1024,
+				CoreNum:    1,
+			}, false)).Should(Equal("256M"))
+
 			Expect(calMysqlPoolSizeByResource(&ResourceDefinition{
 				MemorySize: 2 * 1024 * 1024 * 1024,
 				CoreNum:    2,
-			}, false)).Should(Equal("256M"))
+			}, false)).Should(Equal("384M"))
 
 			// for shard
+			Expect(calMysqlPoolSizeByResource(&ResourceDefinition{
+				MemorySize: 1024 * 1024 * 0.5,
+				CoreNum:    1,
+			}, true)).Should(Equal("128M"))
+
+			Expect(calMysqlPoolSizeByResource(&ResourceDefinition{
+				MemorySize: 1024 * 1024 * 1024,
+				CoreNum:    1,
+			}, true)).Should(Equal("512M"))
```

Comment:
The memory is only 1G and the pool buffer uses 512M? 
Maybe it’s easy to oom?

Reply:
if the machine is shared, the system memory already be  allocate.

Reply:
2g2c only allocates 384M?